flux-typeck/src/constraint_gen.rs

```diff
@@ -220,6 +220,12 @@ fn subtyping(genv: &GlobalEnv, constr: &mut ConstrBuilder, ty1: &Ty, ty2: &Ty, t
                 subtyping(genv, constr, ty1, ty2, tag);
             }
         }
+
+        (_, TyKind::Constr(p2, ty2)) => {
```

Comment:
I think we can also have `Constr` on the left when recursively checking subtyping on type arguments. The following should work for example:

```
Option<{i32[a] : a > 0}> <: Option<i32{v : v > 0 }>
```

 To implement this we need to expose `assume_pred` on `ConstrBuilder`

Reply:
Yes, this is also easy BUT I couldn't think of a way to actually trigger such a subtyping via a test so had left it for now

Reply:
ah figured it out. Anyways, you should **closely review** the code for `constr.push_guard(p)` in the updated PR!